File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/event/UpcomingEventsFragment.java
package com.github.dedis.popstellar.ui.lao.event;

import android.os.Bundle;
import android.view.*;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.github.dedis.popstellar.R;
import com.github.dedis.popstellar.databinding.UpcomingEventsFragmentBinding;
import com.github.dedis.popstellar.ui.lao.LaoActivity;
import com.github.dedis.popstellar.ui.lao.LaoViewModel;
import com.github.dedis.popstellar.ui.lao.event.eventlist.UpcomingEventsAdapter;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class UpcomingEventsFragment extends Fragment {

  private static final Logger logger = LogManager.getLogger(UpcomingEventsFragment.class);
  private LaoViewModel laoViewModel;

  public static UpcomingEventsFragment newInstance() {
    return new UpcomingEventsFragment();
  }

  @Nullable
  @Override
  public View onCreateView(
      @NonNull LayoutInflater inflater,
      @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {

    UpcomingEventsFragmentBinding binding =
        UpcomingEventsFragmentBinding.inflate(inflater, container, false);
    laoViewModel = LaoActivity.obtainViewModel(requireActivity());
    EventsViewModel eventsViewModel =
        LaoActivity.obtainEventsEventsViewModel(requireActivity(), laoViewModel.getLaoId());

    binding.upcomingEventsRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
    binding.upcomingEventsRecyclerView.setAdapter(
        new UpcomingEventsAdapter(
            eventsViewModel.getEvents(), laoViewModel, requireActivity(), logger));

    handleBackNav();
    return binding.getRoot();
  }

  @Override
  public void onResume() {
    super.onResume();
    laoViewModel.setPageTitle(R.string.future_header_title);
    laoViewModel.setIsTab(false);
  }

  private void handleBackNav() {
    LaoActivity.addBackNavigationCallbackToEvents(
        requireActivity(), getViewLifecycleOwner(), logger);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/event/UpcomingEventsFragment.kt
package com.github.dedis.popstellar.ui.lao.event

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.github.dedis.popstellar.R
import com.github.dedis.popstellar.databinding.UpcomingEventsFragmentBinding
import com.github.dedis.popstellar.ui.lao.LaoActivity.Companion.addBackNavigationCallbackToEvents
import com.github.dedis.popstellar.ui.lao.LaoActivity.Companion.obtainEventsEventsViewModel
import com.github.dedis.popstellar.ui.lao.LaoActivity.Companion.obtainViewModel
import com.github.dedis.popstellar.ui.lao.LaoViewModel
import com.github.dedis.popstellar.ui.lao.event.eventlist.UpcomingEventsAdapter

class UpcomingEventsFragment : Fragment() {
  private lateinit var laoViewModel: LaoViewModel

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    savedInstanceState: Bundle?
  ): View {
    val binding = UpcomingEventsFragmentBinding.inflate(inflater, container, false)

    laoViewModel = obtainViewModel(requireActivity())
    val eventsViewModel = obtainEventsEventsViewModel(requireActivity(), laoViewModel.laoId!!)

    binding.upcomingEventsRecyclerView.layoutManager = LinearLayoutManager(context)
    binding.upcomingEventsRecyclerView.adapter =
      UpcomingEventsAdapter(eventsViewModel.events, laoViewModel, requireActivity(), TAG)
    handleBackNav()
    return binding.root
  }

  override fun onResume() {
    super.onResume()
    laoViewModel.setPageTitle(R.string.future_header_title)
    laoViewModel.setIsTab(false)
  }

  private fun handleBackNav() {
    addBackNavigationCallbackToEvents(requireActivity(), viewLifecycleOwner, TAG)
  }

  companion object {
    private val TAG = UpcomingEventsFragment::class.java.simpleName

    @JvmStatic
    fun newInstance(): UpcomingEventsFragment {
      return UpcomingEventsFragment()
    }
  }
}
